src/main/java/uk/ac/cam/cl/dtg/segue/api/AdminFacade.java

```diff
@@ -281,6 +294,114 @@ public synchronized Response modifyUsersRole(@Context final HttpServletRequest r
     return Response.ok().build();
   }
 
+  /**
+   * This method will allow users to have their teacher pending status mass changed.
+   *
+   * @param request - to help determine access rights.
+   * @param status  - new teacher pending status.
+   * @param userIds - a list of user ids to change en-mass
+   * @return Success shown by returning an ok response
+   */
+  @POST
+  @Path("/users/change_teacher_pending/{status}")
+  @Produces(MediaType.APPLICATION_JSON)
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Operation(summary = "Update teacher_pending to true or false for a list of possible user IDs.",
+      description = "This endpoint requires an admin user to be logged in. If updating teacher_pending to false it "
+          + "also sends an email to decline teacher account upgrade. If email send fails, response is still OK but "
+          + "with added message to notify user")
+  public synchronized Response modifyUsersTeacherPendingStatus(@Context final HttpServletRequest request,
+                                                               @PathParam("status") final Boolean status,
```

Comment:
Does this one have to be `Boolean` or can it be `boolean`? You've changed it to `boolean` downstream, which suggests that `null` values are not allowed. If that's the case, I would change this one to `boolean`, unless the framework requires it to be `Boolean`.